pages/api/posts.js

```diff
@@ -1,25 +1,113 @@
-import posts from '../../server/posts-data.json';
+import { readFileSync, writeFileSync } from 'fs';
+import { v4 } from 'uuid';
+import { resolve } from 'path';
+
+const postsDataPath = resolve(__dirname, '../../../../server/posts-data.json');
+
+const postString = readFileSync(postsDataPath).toString();
+let posts = JSON.parse(postString);
+
+function syncPostFile() {
+  writeFileSync(postsDataPath, JSON.stringify(posts));
+}
 
 const PAGE_SIZE = 20;
 
 export default function handler(req, res) {
-  const { postId, pageNumber } = req.query;
-  if (postId) {
-    const post = posts.find((p) => p.post_id === postId);
-    if (post) {
-      res.status(200).json(post);
-    } else {
-      res.status(404).send('Not found.');
+  if (req.method === 'POST') {
+    const { action } = req.query;
+    const { postId, like, commentId, text, replyId } = req.body;
+    console.log('omar1', { postId, like, commentId, text, replyId });
+
+    if (!postId) {
+      return res.status(400).send('Bad request. postId needs to be provided');
+    }
+
+    switch (action) {
+      case 'postLike': {
+        const post = posts.find((p) => p.post_id === postId);
+        post.is_post_liked = like;
+        syncPostFile();
+        return res.status(200).json({
+          ok: true,
+          message: `post is now ${post.is_post_liked ? 'liked' : 'unliked'}`,
+        });
+      }
+      case 'toggleCommentLike': {
+        const post = posts.find((p) => p.post_id === postId);
+        const comment = post.comments.find(
+          (comment) => comment.comment_id === commentId
+        );
+        comment.is_liked_by_user = !comment.is_liked_by_user;
+        syncPostFile();
+        return res.status(200).json({
+          ok: true,
+          message: `comment is now ${
+            comment.is_liked_by_user ? 'liked' : 'unliked'
+          }`,
+        });
+      }
+      case 'toggleReplyLike': {
+        try {
+          const post = posts.find((p) => p.post_id === postId);
+          const comment = post.comments.find(
+            (comment) => comment.comment_id === commentId
+          );
+          const reply = comment.replies.find(
+            (reply) => reply.comment_id === replyId
+          );
+          console.log('omar1');
+          reply.is_liked_by_user = !reply.is_liked_by_user;
+          syncPostFile();
+          return res.status(200).json({
+            ok: true,
+            message: `reply is now ${
+              reply.is_liked_by_user ? 'liked' : 'unliked'
+            }`,
+          });
+        } catch (e) {
+          console.log('omar',  e);
+        }
+      }
+      case 'submitPostComment': {
+        const post = posts.find((p) => p.post_id === postId);
+        post.comments.push({
+          comment_id: v4(),
+          user_name: 'Marwan Al Shaker',
```

Comment:
Usernames can't have spaces. You should use `LOGGED_IN_USER` from [here](https://github.com/alshakerM/my-instagram/blob/add/persistent-updates/stores/constants.js#L1).